client/src/components/CourseResultsView.tsx

```diff
@@ -40,43 +47,45 @@ const CourseResultsView = () => {
     }
   }, [snackPack, messageInfo, alertOpen]);
 
-  const sleep = ms => new Promise(r => setTimeout(r, ms));
+  const sleep = (ms: number) => new Promise(r => setTimeout(r, ms));
 
   // Triggers the calculation of final grades
   const calculateFinalGrades = async () => {
-    setSnackPack((prev) => [...prev,
-      { msg: 'Calculating final grades...', severity: 'info' }
-    ]);
-    await sleep(3000);
-    // Throw error if no grades have been added
-    if (students.length === 0) {
+    try {
       setSnackPack((prev) => [...prev,
-        { msg: 'Import student grades before calculating the final grade.', severity: 'error' }
+        { msg: 'Calculating final grades...', severity: 'info' }
       ]);
-    } else {
-      try {
-        // TODO: connect to backend
+      await sleep(2000);
```

Comment:
What is the purpose of these `sleep` calls?

Reply:
I think the original idea was that the previous snackbar would not be replaced so quickly with other (user has time to read before text changes). In the future this should be updated so that the snackbar can be stacked instead of replaced so there would be no need for this kind of sleep timer.